xsuite/src/proxy/csproxy.ts

```diff
@@ -0,0 +1,171 @@
+import { addressToBechAddress } from '../data/address';
+import { BroadTx, Proxy, unrawTxRes } from './proxy';
+import { e, eCodeMetadata, EncodableAccount } from '../data/encoding';
+import { Kvs } from '../data/kvs';
+
+export class CSProxy extends Proxy {
+  stopChainSimulator: () => void;
+  autoGenerateBlocks: boolean;
+  verbose: boolean;
+
+  constructor(baseUrl: string, stopChainSimulator: () => void, autoGenerateBlocks: boolean = true, verbose: boolean = false) {
+    super(baseUrl);
+
+    this.stopChainSimulator = stopChainSimulator;
+    this.autoGenerateBlocks = autoGenerateBlocks;
+    this.verbose = verbose;
+  }
+
+  static async setAccount(baseUrl: string, account: EncodableAccount, autoGenerateBlocks: boolean = true, verbose: boolean = false) {
+    const [previousAccount, previousKvs] = await Promise.all([
+      CSProxy.getAccount(baseUrl, account.address),
+      CSProxy.getAccountKvs(baseUrl, account.address),
+    ]);
+    const newAccount = accountToRawAccount(account, previousAccount as any, previousKvs as Kvs);

```

Comment:
Because chain simulator overrides the previous state when we call `/set-state`, hence we need to concatenate the previous state with what was changed in the new state.

Reply:
Yes, same in SWorld / simulnet. The setState should receive all the data of the account, and not fetch it inside.

Reply:
Actually no, the setState in simulnet works with partial data as well, eg: you can set only the balance for example without impacting the kvs or the code.

Chain Simulator does not work like that and if we only send the balance we would lose all the other data like the kvs, code etc, hence we need this logic here.

Reply:
You were right, the logic here was not correct and was not the same as in sproxy. The reason being that the tests were not properly setup and to make them work this was needed.

I have now fixed this, the only think that is needed is to fetch the previousKvs and set them to empty if they no longer exist on the account, since the Chain Simulator appends the new keys to the old ones and does not override them.